src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/NewAzureVMDataDiskCommand.cs

```diff
@@ -0,0 +1,153 @@
+        [ValidateNotNullOrEmpty]
+        public string SourceImageUri { get; set; }
+
+        [Parameter(
+            ParameterSetName = ManagedDiskParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMManagedDiskId)]
+        [ValidateNotNullOrEmpty]
+        public string ManagedDiskId { get; set; }
+
+        [Parameter(
+            ParameterSetName = ManagedDiskParameterSet,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = HelpMessages.VMManagedDiskAccountType)]
+        [ValidateNotNullOrEmpty]
+        [PSArgumentCompleter("Standard_LRS", "Premium_LRS")]
+        public string StorageAccountType { get; set; }
+
+        [Parameter(
+            ParameterSetName = ManagedDiskParameterSet,
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = false)]
+        public SwitchParameter WriteAccelerator { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (!string.IsNullOrEmpty(this.Name) && !string.IsNullOrEmpty(this.ManagedDiskId))
+            {
+                if (!this.Name.Equals(GetDiskNameFromId(this.ManagedDiskId)))
+                {
+                    ThrowInvalidArgumentError("Disk name, {0}, does not match with given managed disk ID", this.Name);
+                }
+            }
+
+            var dataDisk = (this.ParameterSetName.Equals(NormalDiskParameterSet))
+                ? new PSVirtualMachineDataDisk
+                {
+                    Name = this.Name,
+                    Caching = this.Caching,
+                    DiskSizeGB = this.DiskSizeInGB,
+                    Lun = this.Lun,
+                    Vhd = string.IsNullOrEmpty(this.VhdUri) ? null : new VirtualHardDisk
+                    {
+                        Uri = this.VhdUri
+                    },
+                    CreateOption = this.CreateOption,
+                    Image = string.IsNullOrEmpty(this.SourceImageUri) ? null : new VirtualHardDisk
+                    {
+                        Uri = this.SourceImageUri
+                    }
+                }
+                : new PSVirtualMachineDataDisk
+                {
+                    Name = this.Name,
+                    Caching = this.Caching,
+                    DiskSizeGB = this.DiskSizeInGB,
+                    Lun = this.Lun,
+                    CreateOption = this.CreateOption,
+                    ManagedDisk = (this.ManagedDiskId == null && this.StorageAccountType == null)
+                              ? null
+                              : new ManagedDiskParameters
+                              {
+                                  Id = this.ManagedDiskId,
+                                  StorageAccountType = this.StorageAccountType
+                              },
+                    WriteAcceleratorEnabled = this.WriteAccelerator.IsPresent
+                };
+
+            WriteObject(dataDisk);
```

Comment:
@hyonholee since `Name` and `ManagedDiskId` are optional parameters, there are scenarios where the disk that is being returned doesn't have a name or managed disk -- how is this handled when you try to add the disk to a VMSS VM and update it on the server?